cdk/lib/constructs/worker/image-builder.ts

```diff
@@ -115,13 +118,33 @@ export class WorkerImageBuilder extends Construct {
         parameters: {
           imagePipelineArn: pipeline.pipeline.attrArn,
         },
-        physicalResourceId: PhysicalResourceId.of(recipeVersion.getAttString('version')),
+        physicalResourceId: PhysicalResourceId.of(`${recipeVersion.getAttString('version')}#${amiVersion}`),
       },
       policy: AwsCustomResourcePolicy.fromSdkCalls({
         resources: [pipeline.pipeline.attrArn],
       }),
     });
 
+    new AwsCustomResource(this, 'PurgeAmiCache', {
+      onUpdate: {
+        service: '@aws-sdk/client-ssm',
+        action: 'DeleteParameter',
+        parameters: {
+          Name: props.amiIdParameterName,
+        },
+        ignoreErrorCodesMatching: 'ParameterNotFound',
+        physicalResourceId: PhysicalResourceId.of(amiVersion),
```

Comment:
We allow to force AMI id parameter now, because when dynamodb table is replaced, it throws error on worker started from old AMI.